base-template.sh

```diff
@@ -89,8 +95,11 @@ load_install_dir() {
 # Returns: None
 #####################################################
 set_main_variables() {
-    HOOK_NAME=$(basename "$0")
-    HOOK_FOLDER=$(dirname "$0")
+
+    HOOK_NAME=$(basename "$1")
+    HOOK_FOLDER=$(dirname "$1")
+    INSTALL_DIR="$2"
```

Comment:
Is it really needed to pass INSTALL_DIR here? It slightly complicates the symlink-to-base-template interface and the symlink script and might introduce a chance of mismatch? Did you consider the "derive from $0-trick` I suggested [originally](https://github.com/rycus86/githooks/pull/92#discussion_r424998436)?

Reply:
Do you know if `$0` is POSIX and expected to work under any shells?

Reply:
https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_05_02

> 0    (Zero.) Expands to the name of the shell or shell script. See sh for a detailed description of how this name is derived.

Then the sh page has more details:

https://pubs.opengroup.org/onlinepubs/009695399/utilities/sh.html

Here, it's relevant how `sh` was invoked. WHen using `#!/bin/sh` AFAIU the kernel just appends the script filename that contains the `#!` to `/bin/sh`, so it invokes this flavor of the `sh` commandline:

    sh [-abCefhimnuvx][-o option][+abCefhimnuvx][+o option] [command_file [argument...]]

So then, this rule triggers:

> Special parameter 0 (see Special Parameters) shall be set to the value of command_file. If sh is called using a synopsis form that omits command_file, special parameter 0 shall be set to the value of the first argument passed to sh from its parent (for example, argv[0] for a C program), which is normally a pathname used to execute the sh utility.

IOW, `$0` is always set to the value of the script for cases we care about (i.e. when using the `#!`, or when just running `sh /path/to/base-template.sh`. There is a way to let `$0` point to something else (by specifying it explicitly in the ` sh -c[-abCefhimnuvx][-o option][+abCefhimnuvx][+o option]command_string [command_name [argument...]]` form), but that is not applicable here.

Reply:
Thanks for checking!